bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/codecs/Codecs.java

```diff
@@ -24,17 +24,44 @@
 import org.eclipse.emfcloud.modelserver.common.codecs.XmiCodec;
 
 import com.fasterxml.jackson.databind.JsonNode;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
 
 import io.javalin.http.Context;
 import io.javalin.websocket.WsContext;
 
 public class Codecs {
 
+   /** The constant to tell which is the preferred formats in codecs. */
+   public static final String PREFERRED_FORMAT = "PreferredFormat";
+
+   @Inject(optional = true)
+   @Named(PREFERRED_FORMAT)
+   private final String preferredFormat = ModelServerPathParameters.FORMAT_JSON;
+
    private final Map<String, Codec> formatToCodec = new LinkedHashMap<>();
 
+   /**
+    * Legacy constructor in case users were instanciating Codecs class manually.
```

Comment:
minor typo: instanciating -> instantiating

Reply:
Apparently quite a common typo when you google it. :+1: